pkgs/servers/gpsd/default.nix

```diff
@@ -81,6 +81,7 @@ stdenv.mkDerivation rec {
     patchShebangs .
     sed -e "s|systemd_dir = .*|systemd_dir = '$out/lib/systemd/system'|" -i SConscript
     export TAR=noop
+    substituteInPlace SConscript --replace "env['CCVERSION']" "env['CC']"
```

Comment:
I wonder why Buildroot doesn't need this fix: https://git.buildroot.org/buildroot/tree/package/gpsd/gpsd.mk

Reply:
> I wonder why Buildroot doesn't need this fix: https://git.buildroot.org/buildroot/tree/package/gpsd/gpsd.mk

Do they apply the same `sconstruct-env-fixes.patch` that we do?  I'm not familiar with buildroot but from a quick glance at that link it doesn't look like it.


> ofborg build pkgsCross.mips64el-linux-gnuabi64.gpsd

```
error: 4 dependencies of derivation '/nix/store/6snyz4aq6cq5i80ygsqzg8qb8w32973q-gpsd-mips64el-unknown-linux-gnuabi64-3.23.1.drv' failed to build
```

Oh right, ofborg doesn't have my `guiSupport=false` overlay, so it tries to build `pkgsCross.mips64el-linux-gnuabi64.gtk`, which isn't working yet.  Let's try:

@ofborg build pkgsCross.aarch64-multiplatform.gpsd

Reply:
> @ofborg build pkgsCross.aarch64-multiplatform.gpsd

Hrm I guess I have to say that in a top-level comment.

Reply:
Looks like it is a [bug in SCons](https://github.com/SCons/scons/issues/1723) that is only triggered if you set the `CC` environment variable.  I bet buildroot just makes sure that `${multiarch-name}/bin/` is in the `$PATH`.

Although the bug is closed there is no mention to what fixed it.  I don't think it's actually fixed.  It looks like they imported a mass of bugs from some non-github bug tracking system.

A lot of people are experiencing this when setting the `CC` environment variable:

https://github.com/SCons/scons/issues?q=is%3Aissue+ccversion+is%3Aclosed

Reply:
Nice find!

Reply:
I don't know how many SCons-based packages we have in nixpkgs, but it seems like that CCVERSION/CC environment variable fix is applicable to SCons itself and not this gpsd package in particular? Should we turn that fix into a scons setupHook instead?

Reply:
The `SConscript` is part of the `gpsd` source tarball, so no.

Reply:
I'm not convinced by that argument because the cmake setupHook patches CMakeLists.txt files in upstream source tarballs -- isn't this the same kind of situation?

Reply:
No, it is not.  The problematic line was added to the SConscript in [this](https://gitlab.com/gpsd/gpsd/-/commit/d0558636038e18d90eb886f668ff5004538f7839) commit to the gpsd repository by a gpsd maintainer.

Reply:
All [`fixCmakeFiles()`](https://github.com/NixOS/nixpkgs/blob/57b2048cc8c34f9a2e3a2edc196e9a98efc3862f/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L6) does is eliminate FHS-assuming absolute paths (`/usr/..`, etc).  It's basically the cmake version of `patchShebangs`.

Reply:
Correct me if I'm wrong, but isn't the problem here that we trigger a scons bug by setting `$CC`, which means that scons fails to set `$CCVERSION`? IOW, if we had a setupHook in scons that populated `$CCVERSION` then we didn't need to patch gpsd?

Anyway, I don't feel _that_ strongly about it, but it would be nice to either fix it upstream in scons or work around generically in nixpkgs (with scons setupHook).

Reply:
Looks like scons _intends_ to set `$CCVERSION` when using clang, gcc and aixcc (search for `CCVERSION` here: https://scons.org/doc/production/HTML/scons-man.html#cv-CCVERSION). Specifically, the `$CCVERSION` env var _may_ be set, depending on which compiler is used.

I think we can say that gpsd shouldn't rely on `$CCVERSION` being set, although it's clearly a scons bug that it doesn't get set if `$CC` is set.

Reply:
Can you please update the commit message with a summary of what we know now?

Reply:
> Can you please update the commit message with a summary of what we know now?

This needs to be resolved in order to merge.

Reply:
I am sorry, I cannot figure out if the commit message was updated or not, as requested.